server/routes/gcp/billing.py

```
@@ -18,49 +19,51 @@ def billing(user_id):
+                projects = get_project_list(credentials) or []
             except Exception as e:
-                logging.warning(f"Error checking project {project_id}: {e}. Skipping.")
+                logging.warning("Failed to list projects for sa=%s: %s", sa_label[:12], type(e).__name__)
                 continue
+            for project in projects:
+                project_id = project.get('projectId')
+                if not project_id or project_id in processed_projects:
+                    continue
+                try:
+                    if is_bigquery_enabled(project_id, credentials):
+                        logging.info(f"Storing billing data for project: {project_id}")
```

Ruff (0.15.14)

[warning] Logging statement uses f-string

(G004)